hw_3/script.js

```diff
@@ -0,0 +1,111 @@
+function checkArgLength(args) {
+    if (args.length !== 2) {
+        throw new Error("Not 2 arguments");
+    }
+}
+
+function checkMinMax(min, max) {
+    if (!Number.isInteger(min) || !Number.isInteger(max)) {
+        throw new Error("arguments are not Integer");
+    }
+    if (min > max) {
+        throw new Error("Min > Max");
+    }
+}
+
+function checkResult(result) {
+    if (result > Number.MAX_SAFE_INTEGER) {
+        return new Error("The result is greater than Number.MAX_SAFE_INTEGER")
+    } else {
+        return result;
+    }
+
+}
+
+function range(min, max) {
```

Comment:
What do you think about adding the checker functions directly here, without creating little functions and then calling a function that calls a bunch of other functions? 
I'm not sure here, but it looks better to me.
```suggestion
function range(min, max) {
    if (arguments.length !== 2) {
        throw new Error("Function has a different number of arguments than two");
    }

    if (!Number.isInteger(min) || !Number.isInteger(max)) {
        throw new Error("Arguments (min, max) are not Integer");
    }

    if (min > max) {
        throw new Error("Minimum value (min) is greater than maximum value (max)");
    }

    return checkValueForMaxSafeInteger((min + max) * (max - min + 1) / 2);
}
```

Reply:
Moved error checking to range function.
https://github.com/BlackTruth/javascript/blob/813db56a38899f41dcfdc4b332943bde5c271592/hw_3/script.js#L9-L21